pkg/util/clusterdata/ingress_profiles_task.go

```diff
@@ -0,0 +1,116 @@
+package clusterdata
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+func newIngressProfileEnricherTask(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftCluster) (enricherTask, error) {
+	operatorcli, err := operatorclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kubecli, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ingressProfileEnricherTask{
+		log:         log,
+		operatorcli: operatorcli,
+		kubecli:     kubecli,
+		oc:          oc,
+	}, nil
+}
+
+type ingressProfileEnricherTask struct {
+	log         *logrus.Entry
+	operatorcli operatorclient.Interface
+	kubecli     kubernetes.Interface
+	oc          *api.OpenShiftCluster
+}
+
+func (ef *ingressProfileEnricherTask) FetchData(ctx context.Context, callbacks chan<- func(), errs chan<- error) {
+	//List IngressControllers from  openshift-ingress-operator namespace
```

Comment:
Nit: `// Comment` We usually leave empty scale before comment. Makes easier to read